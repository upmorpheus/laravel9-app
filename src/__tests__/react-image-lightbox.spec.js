/* eslint-env jest */
import { shallow } from 'enzyme';
import React from 'react';
import Modal from 'react-modal';
import Lightbox from '../index';

// Mock the loadStyles static function to avoid
// issues with a lack of styles._insertCss
Lightbox.loadStyles = jest.fn();

const commonProps = {
  mainSrc: '/fake/image/src.jpg',
  onCloseRequest: () => {},
};

describe('With Enzyme', () => {
  it('Lightbox has modal', () => {
    const wrapper = shallow(<Lightbox {...commonProps} />);
    expect(wrapper.find(Modal).length).toEqual(1);
  });
});

describe('With Snapshot Testing', () => {
  it('Lightbox renders properly"', () => {
    const wrapper = shallow(<Lightbox {...commonProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
