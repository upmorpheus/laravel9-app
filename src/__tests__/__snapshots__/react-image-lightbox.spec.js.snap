// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Lightbox renders properly" 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentLabel="Lightbox"
  isOpen={true}
  onAfterOpen={[Function]}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "backgroundColor": "transparent",
        "border": "none",
        "borderRadius": 0,
        "bottom": 0,
        "left": 0,
        "overflow": "hidden",
        "padding": 0,
        "right": 0,
        "top": 0,
      },
      "overlay": Object {
        "backgroundColor": "transparent",
        "zIndex": 1000,
      },
    }
  }
>
  <div
    className="ril-outer outer outerAnimating  "
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    onWheel={[Function]}
    style={
      Object {
        "animationDirection": "reverse",
        "animationDuration": "300ms",
        "transition": "opacity 300ms",
      }
    }
    tabIndex="-1"
  >
    <div
      className="ril-inner inner"
      onClick={[Function]}
    >
      <img
        alt="Image"
        className="ril-image-current image"
        draggable={false}
        key="/fake/image/src.jpgi0"
        onDoubleClick={[Function]}
        onDragStart={[Function]}
        onWheel={[Function]}
        src="/fake/image/src.jpg"
        style={
          Object {
            "transform": "translate3d(0px,0px,0) scale3d(NaN,NaN,1)",
          }
        }
      />
    </div>
    <div
      className="ril-toolbar toolbar"
    >
      <ul
        className="ril-toolbar-left toolbarSide toolbarLeftSide"
      >
        <li
          className="ril-toolbar__item toolbarItem"
        >
          <span
            className="ril-toolbar__item__child toolbarItemChild"
          />
        </li>
      </ul>
      <ul
        className="ril-toolbar-right toolbarSide toolbarRightSide"
      >
        <li
          className="ril-toolbar__item toolbarItem"
        >
          <button
            aria-label="Zoom in"
            className="ril-zoom-in toolbarItemChild builtinButton zoomInButton"
            key="zoom-in"
            onClick={[Function]}
            type="button"
          />
        </li>
        <li
          className="ril-toolbar__item toolbarItem"
        >
          <button
            aria-label="Zoom out"
            className="ril-zoom-out toolbarItemChild builtinButton zoomOutButton builtinButtonDisabled"
            key="zoom-out"
            onClick={[Function]}
            type="button"
          />
        </li>
        <li
          className="ril-toolbar__item toolbarItem"
        >
          <button
            aria-label="Close lightbox"
            className="ril-close ril-toolbar__item__child toolbarItemChild builtinButton closeButton"
            key="close"
            onClick={[Function]}
            type="button"
          />
        </li>
      </ul>
    </div>
  </div>
</Modal>
`;
